nuxeo-core/nuxeo-core-event/src/main/java/org/nuxeo/ecm/core/work/StreamWorkManager.java

```diff
@@ -471,8 +489,8 @@ protected long getLowWaterMark(String queueId) {
 
     @Override
     public Work.State getWorkState(String s) {
-        // always not found
-        return null;
+        KeyValueStore kvStore = Framework.getService(KeyValueService.class).getKeyValueStore(KV_NAME);
```

Comment:
What do you think to raise an exception if store state is not enabled ? Maybe it's too strict and will break stuff in Nuxeo... If it's case, what do you think to not call KV if `storeState` is `false` ?

Reply:
Good point, I've done it already in an upcoming commit for the ticket https://jira.nuxeo.com/browse/NXP-24400

Reply:
👍 
I've "backported" it to this branch